test/integration/e2e_test.go

```diff
@@ -290,14 +290,28 @@ func TestE2EFullIntegration(t *testing.T) { // nolint:gocyclo
 			Resource: "namespaces",
 		})
 
+		var additionalScopes []string
+		// If we're using dex, we will test that we see a warning when the access token
+		// lifetime is too short (we have it set to 20 minutes) and it's using access token based refresh.
+		// To ensure that access token refresh happens rather than refresh token, don't ask for the offline_access scope.
+		// In other environments, test the refresh token based flow.
```

Comment:
Er wouldn't it be better to have a separate test that does the access token based test and looks for the warning?